### PR TITLE
On chain inbox

### PR DESCRIPTION
**Pull request by hkalodner**
Switch to storing all inbox messages on chain rather than just an accumulator

### CODE REVIEW

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by edfelten — CHANGES_REQUESTED** [review]
Just one small issue in a comment.  Otherwise looks good.

File: packages/arb-bridge-eth/contracts/rollup/Rollup.sol [hunk 1]
```diff
@@ -471,7 +470,7 @@ contract Rollup is Cloneable, RollupCore, Pausable, IRollup {
         );
 
         // inboxMaxCount must be greater than beforeInboxCount since we can't have read past the end of the inbox
```
[the remark this answers is not in the document]
Comment (hkalodner) [reply packages/arb-bridge-eth/contracts/rollup/Rollup.sol]: Updated comment

**Review by edfelten — APPROVED** [review]
Looks good.

**Pull request merged**